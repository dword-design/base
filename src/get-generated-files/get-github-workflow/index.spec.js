import P from 'node:path';

import chdir from '@dword-design/chdir';
import tester from '@dword-design/tester';
import testerPluginTmpDir from '@dword-design/tester-plugin-tmp-dir';
import { execaCommand } from 'execa';
import fs from 'fs-extra';
import isCI from 'is-ci';
import outputFiles from 'output-files';

import { Base } from '@/src/index.js';

export default tester(
  {
    'GitHub CLI exists': async () => {
      if (isCI) {
        await execaCommand('gh repo list');
      }
    },
    async 'job matrix'() {
      await fs.outputFile('package.json', JSON.stringify({}));

      expect(
        new Base({ useJobMatrix: true }).getGithubWorkflowConfig(),
      ).toMatchSnapshot(this);
    },
    async 'job matrix no macos'() {
      await fs.outputFile('package.json', JSON.stringify({}));

      expect(
        new Base({
          macos: false,
          useJobMatrix: true,
        }).getGithubWorkflowConfig(),
      ).toMatchSnapshot(this);
    },
    async 'job matrix no windows'() {
      await fs.outputFile('package.json', JSON.stringify({}));

      expect(
        new Base({
          useJobMatrix: true,
          windows: false,
        }).getGithubWorkflowConfig(),
      ).toMatchSnapshot(this);
    },
    async 'no job matrix'() {
      await fs.outputFile('package.json', JSON.stringify({}));
      expect(new Base({}).getGithubWorkflowConfig()).toMatchSnapshot(this);
    },
    async 'package.json'() {
      await outputFiles({
        '.env.schema.json': JSON.stringify({ foo: { type: 'string' } }),
        'repos/foo/package.json': JSON.stringify({}),
      });

      await chdir(P.join('repos', 'foo'), () =>
        expect(new Base().getGithubWorkflowConfig()).toMatchSnapshot(this),
      );
    },
    async 'package.json same path as .env.schema.json'() {
      await outputFiles({
        'repos/foo': {
          '.env.schema.json': JSON.stringify({ foo: { type: 'string' } }),
          'package.json': JSON.stringify({}),
        },
      });

      await chdir(P.join('repos', 'foo'), () =>
        expect(new Base().getGithubWorkflowConfig()).toMatchSnapshot(this),
      );
    },
    async 'test environment variables'() {
      await outputFiles({
        '.env.schema.json': JSON.stringify({ bar: {}, foo: {} }),
        'package.json': JSON.stringify({}),
      });

      expect(
        new Base({
          nodeVersion: 14,
          useJobMatrix: false,
        }).getGithubWorkflowConfig(),
      ).toMatchSnapshot(this);
    },
    async testInContainer() {
      await fs.outputFile('package.json', JSON.stringify({}));

      expect(
        new Base({
          nodeVersion: 14,
          testInContainer: true,
          useJobMatrix: true,
        }).getGithubWorkflowConfig(),
      ).toMatchSnapshot(this);
    },
  },
  [testerPluginTmpDir()],
);
