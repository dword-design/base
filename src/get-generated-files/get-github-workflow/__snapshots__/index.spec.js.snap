// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index job matrix 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "run": "pnpm lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 20",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 18,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "macos-latest",
            },
            Object {
              "node": 20,
              "os": "windows-latest",
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index job matrix no macos 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "run": "pnpm lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 20",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 18,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "windows-latest",
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index job matrix no windows 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "run": "pnpm lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 20",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 18,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "macos-latest",
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index no job matrix 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index package.json 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "run": "pnpm lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 20",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 18,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "macos-latest",
            },
            Object {
              "node": 20,
              "os": "windows-latest",
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index package.json same path as .env.schema.json 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 20,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "run": "pnpm lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "TEST_FOO": "\${{ secrets.TEST_FOO }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 20",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 18,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 20,
              "os": "macos-latest",
            },
            Object {
              "node": 20,
              "os": "windows-latest",
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index subdir 1`] = `
Object {
  "jobs": Object {
    "cancel-existing": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
      ],
    },
    "release": Object {
      "needs": "test",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": 14,
          },
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn checkUnknownFiles",
        },
        Object {
          "run": "yarn lint",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "yarn dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "NPM_TOKEN": "\${{ secrets.NPM_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "yarn semantic-release",
        },
      ],
    },
    "test": Object {
      "needs": "cancel-existing",
      "runs-on": "\${{ matrix.os }}",
      "steps": Array [
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "node-version": "\${{ matrix.node }}",
          },
        },
        Object {
          "run": "yarn --frozen-lockfile",
        },
        Object {
          "run": "yarn test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v3",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "if": "matrix.os == 'ubuntu-latest' && matrix.node == 14",
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
      ],
      "strategy": Object {
        "matrix": Object {
          "include": Array [
            Object {
              "node": 14,
              "os": "ubuntu-latest",
            },
            Object {
              "node": 16,
              "os": "ubuntu-latest",
            },
            Object {
              "os": "macos-latest",
              "version": 16,
            },
            Object {
              "os": "windows-latest",
              "version": 16,
            },
          ],
        },
      },
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index test environment variables 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": 14,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
            "TEST_BAR": "\${{ secrets.TEST_BAR }}",
            "TEST_FOO": "\${{ secrets.TEST_FOO }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;

exports[`index testInContainer 1`] = `
Object {
  "jobs": Object {
    "build": Object {
      "if": "!contains(github.event.head_commit.message, '[skip ci]')",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "uses": "rokroskar/workflow-run-cleanup-action@v0.3.3",
        },
        Object {
          "uses": "actions/checkout@v4",
          "with": Object {
            "fetch-depth": 0,
            "lfs": true,
            "ref": "\${{ github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.head.ref || '' }}",
          },
        },
        Object {
          "uses": "actions/setup-node@v4",
          "with": Object {
            "check-latest": true,
            "node-version": 14,
          },
        },
        Object {
          "run": "corepack enable",
        },
        Object {
          "run": "git config --global user.email \\"actions@github.com\\"",
        },
        Object {
          "run": "git config --global user.name \\"GitHub Actions\\"",
        },
        Object {
          "run": "pnpm install --frozen-lockfile",
        },
        Object {
          "env": Object {
            "GH_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "run": "pnpm test",
        },
        Object {
          "if": "failure()",
          "uses": "actions/upload-artifact@v4",
          "with": Object {
            "name": "Image Snapshot Diffs",
            "path": "**/__image_snapshots__/__diff_output__",
          },
        },
        Object {
          "uses": "codecov/codecov-action@v3",
          "with": Object {
            "token": "\${{ secrets.CODECOV_TOKEN }}",
          },
        },
        Object {
          "run": "pnpm checkUnknownFiles",
        },
        Object {
          "env": Object {
            "GITHUB_REPOSITORY": "\${{ secrets.GITHUB_REPOSITORY }}",
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Push changed files",
          "run": "pnpm dw-ci push-changed-files",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "if": "github.ref == 'refs/heads/master'",
          "name": "Release",
          "run": "pnpm semantic-release",
        },
      ],
    },
  },
  "name": "build",
  "on": Object {
    "pull_request": Object {},
    "push": Object {
      "branches": Array [
        "master",
      ],
    },
  },
}
`;
